src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.EmptyFieldException;
+import core.basesyntax.exception.IllegalAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.LoginExistingException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final String EMPTY_FIELD_MESSAGE = "Login/password or age is empty. "
+            + "Please, write correct input data to continue!";
+    private static final String INVALID_LOGIN_MESSAGE = "Login %s is incorrect. "
+            + "Your login cannot be less than 6 characters!";
+    private static final String INVALID_PASSWORD_MESSAGE = "Password %s is incorrect. "
+            + "Your password cannot be less than 6 characters!";
+    private static final String INCORRECT_AGE_MESSAGE = "Age %d is incorrect. "
+            + "Age cannot be less than 18 years old or over the %d!";
+    private static final String LOGIN_EXISTS_MESSAGE = "Login %s is already in storage. "
+            + "Please, write another login to continue!";
+    private static final int MAX_PERMISSIBLE_AGE = 122;
+    private static final int MIN_PERMISSIBLE_AGE = 18;
+    private static final int MIN_CHARACTER_AMOUNT = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUserData(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void validateUserData(User user) {
+        String userLogin = user.getLogin();
+        String userPassword = user.getPassword();
+        Integer userAge = user.getAge();
+        if (userLogin == null || userPassword == null || userAge == null) {
```

Comment:
[Common mistakes](https://github.com/proxychang/jv-registration-validation/blob/hw-jv-registration-validation/checklist.md#common-mistakes-jv-registration-validation)

- Don't create one big if with all conditions. Better to split one big if into several small ones.

Bad example:
```
  public User register(User user) {
   if (user.getLogin() == null 
     || user.getPassword() == null
     || user.getAge() < MIN_AGE) {
     throw new RegistrationException("Invalid data");
   }
   return storageDao.add(user);
  }
```
Good example:
```
  public User register(User user) {
   if (user.getLogin() == null) {
     throw new RegistrationException("Login can't be null");
   }
   if (user.getPassword() == null) {
     throw new RegistrationException("Password can't be null");
   }
   if (user.getAge() < MIN_AGE) {
     throw new RegistrationException("Not valid age: " + user.getAge() + ". Min allowed age is " + MIN_AGE);
   }
   return storageDao.add(user);
  }
```